strr-web/pages/registry-dashboard.vue

```diff
@@ -63,45 +63,105 @@
+            <UButton
+              v-if="row.isPaid"
+              :icon="downloadingReceipts.has(row.applicationNumber) ? 'i-mdi-download' : 'i-mdi-receipt-text-outline'"
+              variant="ghost"
+              :disabled="downloadingReceipts.has(row.applicationNumber)"
+              :aria-label="tRegistryDashboard('downloadReceipt')"
+              class="h-[20px] w-[20px] ml-3 !p-0 text-blue-600"
+              @click="downloadReceipt(row.applicationNumber)"
+            />
           </div>
         </template>
         <template #registrationNumber-data="{ row }">
           <div class="flex items-center">
-            <span
-              class="cursor-pointer"
-              @click="
+            <div
+              :class="{ 'cursor-pointer': isClickableRow(row.registrationType) }"
+              class="text-center min-w-[80px]"
+              @click="handleRowClick(row.registrationType, () =>
                 row.registrationId
                   ? navigateToRegistrationDetails(row.registrationId)
                   : navigateToApplicationDetails(row.applicationNumber)
-              "
+              )"
             >
               {{ row.registrationNumber }}
-            </span>
-            <UIcon
-              v-if="row.isCertificateIssued"
-              name="i-mdi-file-document-outline"
-              class="h-[20px] w-[20px] ml-3 text-blue-600 cursor-pointer"
-              alt="certificate download"
-              @click="downloadCertificate(row.registrationId)"
-            />
+            </div>
           </div>
         </template>
         <template #registrationType-data="{ row }">
-          <div class="cursor-pointer w-full" @click="navigateToApplicationDetails(row.applicationNumber)">
+          <div
+            :class="{ 'cursor-pointer': isClickableRow(row.registrationType) }"
+            class="w-full"
+            tabindex="0"
+            @click="handleRowClick(
+              row.registrationType,
+              () => navigateToApplicationDetails(row.applicationNumber)
+            )"
+            @keydown.enter="handleRowClick(
+              row.registrationType,
+              () => navigateToApplicationDetails(row.applicationNumber)
+            )"
+            @keydown.space.prevent="handleRowClick(
+              row.registrationType,
+              () => navigateToApplicationDetails(row.applicationNumber)
+            )"
+          >
             {{ row.registrationType }}
           </div>
         </template>
         <template #address-data="{ row }">
-          <div class="cursor-pointer w-full" @click="navigateToApplicationDetails(row.applicationNumber)">
-            <div>{{ row.unitAddress }}</div>
-            <div>{{ row.unitAddressCity }} {{ row.unitAddressPostalCode }}</div>
+          <div class="flex items-center">
+            <div
+              :class="{ 'cursor-pointer': isClickableRow(row.registrationType) }"
+              tabindex="0"
+              @click="handleRowClick(
```

Comment:
We can cleanup a bit with something like this:
`@click="isClickableRow(row.registrationType) && navigateToApplicationDetails(row.applicationNumber)"` 
or
`@click=handleRowClick(row)` <= without passing callback, since it is the same.

Reply:
Done

Reply:
Thank you, looks good!